src/v/cluster/rm_stm.cc

```diff
@@ -1099,10 +1100,36 @@ void rm_stm::compact_snapshot() {
     if (cutoff_timestamp <= _oldest_session.value()) {
         return;
     }
+
+    if (_log_state.seq_table.size() < _seq_table_min_size) {
+        return;
+    }
+
+    if (_log_state.seq_table.size() == 0) {
+        return;
+    }
+
+    std::vector<model::timestamp::type> lw_tss;
+    lw_tss.reserve(_log_state.seq_table.size());
+    for (auto it = _log_state.seq_table.cbegin();
+         it != _log_state.seq_table.cend();
+         it++) {
+        lw_tss.push_back(it->second.last_write_timestamp);
+    }
+    std::sort(lw_tss.begin(), lw_tss.end());
+    auto pivot = lw_tss[lw_tss.size() - 1 - _seq_table_min_size];
```

Comment:
it looks like if `_log_state.seq_table.size() == _seq_table_min_size` then `lw_tss.size() - 1 - _seq_table_min_size` is `-1` which would be undefined behavior?

Reply:
added a check to prevent it, thanks!